paddle/fluid/recordio/scanner.h

```diff
@@ -0,0 +1,44 @@
+//   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <fstream>
+#include <memory>
+#include "paddle/fluid/recordio/chunk.h"
+namespace paddle {
+namespace recordio {
+
+class Scanner {
+public:
+  explicit Scanner(std::unique_ptr<std::istream>&& stream);
```

Comment:
Since we store the pointer of an istream object, there is no difference between make a right value copy or not.

Reply:
`Scanner` will read the istream until ends. It will steal the ownership of stream and free the stream inside. So, here use right-value reference.

Reply:
BTW, this API is not used for operators. It just used for unit tests.

Reply:
Sorry, I didn't point out my opinion clearly. I thought that we can only use an `std::istream*` as the constructor parameter. Otherwise, it will force other modules to use unique_ptr. Nevermind since it just used for unit tests.